### PR TITLE
Update roadmap to reflect reality.

### PR DESCRIPTION
**Pull request by cpuguy83**
The roadmap is one of the most important ways that a new contributor may
get started on the codebase, as such it is important for it to reflect
the real effort that is currently happening.

This update just brings it up to date. There may be some other efforts
going on and I would encourage people to update the roadmap accordingly
as a separate effort.

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
SGTM :angel:

File: ROADMAP.md
Comment (thaJeztah) [inline ROADMAP.md]: s/functionaility/functionality/

File: ROADMAP.md
Comment (thaJeztah) [inline ROADMAP.md]: `now` is in this sentence twice; could remove one of them

File: ROADMAP.md
Comment (thaJeztah) [inline ROADMAP.md]: s/currnetly/currently/

s/rutime/runtime/

File: ROADMAP.md
Comment (thaJeztah) [inline ROADMAP.md]: s/cheif/chief/ (unless you were referring to [this urban dictionary use of that word](https://www.urbandictionary.com/define.php?term=Cheif) :joy:)

File: ROADMAP.md
Comment (thaJeztah) [reply ROADMAP.md]: Perhaps something like "a custom syntax" (it doesn't have to be a Dockerfile (like) syntax)

**Review by thaJeztah — CHANGES_REQUESTED** [review]

**Comment by thaJeztah** [comment]
Thanks for starting the work on this! I left some nits, and we'll definitely need more follow-ups, but it's a good start 👍

**Comment by thaJeztah** [comment]
As came up in the maintainers meeting, I think the big elephant in the room is also the API; how do we want to evolve the API? Split it to a separate repo (keep the definition/specs separate from implementation)? Head more towards a GRPC API, and keep the RESTful API as an optional compatibility layer?

We can keep that for a follow-up, but it's something we should discuss

**Review by iaoiui — CHANGES_REQUESTED** [review]

File: ROADMAP.md
Comment (cpuguy83) [reply ROADMAP.md]: Can you write what you'd like to see here and I'll update?

**Comment by cpuguy83** [comment]
@AkihiroSuda Added.

**Comment by thaJeztah** [comment]
ping @moby/moby-maintainers PTAL if things are missing that you'd like to have added

**Review by AkihiroSuda — APPROVED** [review]

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

File: ROADMAP.md
Comment (cpuguy83) [reply ROADMAP.md]: @tonistiigi WDYT?

Comment (tonistiigi) [reply ROADMAP.md]: New features for builder and Dockerfile should be implemented first in BuildKit backend using an external Dockerfile implementation from the container images. This allows everyone to test and evaluate the feature without upgrading their daemon. New features should go to experimental channel first and can be part of the `docker/dockerfile:experimental` image. From there they graduate to `docker/dockerfile:latest` and binary releases. The Dockerfile frontend source code is temporarily located at https://github.com/moby/buildkit/tree/master/frontend/dockerfile with separate new features defined with go build tags.

**Comment by AkihiroSuda** [comment]
Mergeable?

File: ROADMAP.md
Comment (cpuguy83) [reply ROADMAP.md]: Done

File: ROADMAP.md
Comment (thaJeztah) [inline ROADMAP.md]: The markdown for the link is flipped should be;

```suggestion
See [buildpacks on buildkit](https://github.com/tonistiigi/buildkit-pack) as an
```

File: ROADMAP.md
Comment (thaJeztah) [inline ROADMAP.md]: Looks like GItHub won't render this as a link; can you make it a link? Perhaps add the title of the issue as well;

```suggestion
- [#38043](https://github.com/moby/moby/issues/38043) Proposal: containerd image integration
```

File: ROADMAP.md
Comment (thaJeztah) [inline ROADMAP.md]: Same for this one;

```suggestion
- [#32925](https://github.com/moby/moby/issues/32925) discussion: builder future: buildki
```

File: ROADMAP.md
Comment (thaJeztah) [inline ROADMAP.md]:
```suggestion
- [#37375](https://github.com/moby/moby/issues/37375) Proposal: allow running `dockerd` as an unprivileged user (aka rootless mode)
```

File: ROADMAP.md
Comment (thaJeztah) [inline ROADMAP.md]:
```suggestion
- [#32866](https://github.com/moby/moby/issues/32866) Replace integration-cli suite with API test suite
```

File: ROADMAP.md
Comment (thaJeztah) [inline ROADMAP.md]:
```suggestion
example of this extensibility.
```

File: ROADMAP.md
Comment (thaJeztah) [inline ROADMAP.md]:
```suggestion
New features for the builder and Dockerfile should be implemented first in the
```

File: ROADMAP.md
Comment (thaJeztah) [inline ROADMAP.md]:
```suggestion
their daemon. New features should go to the experimental channel first, and can be
```

File: ROADMAP.md
Comment (thaJeztah) [inline ROADMAP.md]:
```suggestion
support running the daemon as a nomral, unprivileged user without requiring `suid`
```

**Review by thaJeztah — APPROVED** [review]
Left some minor suggestions and fixes, but LGTM after those are addressed

**Comment by AkihiroSuda** [comment]
@cpuguy83 could you address @thaJeztah 's comments?

**Comment by cpuguy83** [comment]
Updated.

**Comment by olljanat** [comment]
> As came up in the maintainers meeting, I think the big elephant in the room is also the API; how do we want to evolve the API? Split it to a separate repo (keep the definition/specs separate from implementation)?

Based on my experience so far I think that it would be good idea to move at least ```github.com/docker/docker/api/types``` to own repository. Now some changes like this one https://github.com/moby/moby/issues/25885#issuecomment-447657852 needs quite boring PR loop between repositories.

> We can keep that for a follow-up, but it's something we should discuss

Should we create separate issue for that discussion?

**Comment by cpuguy83** [comment]
I don't think it belongs in the roadmap, for sure.

**Comment by olljanat** [comment]
Derek add label: rebuild/janky

**Comment by thaJeztah** [comment]
There's this issue as well, so we can use that for discussing; https://github.com/moby/moby/issues/32873

**Review by thaJeztah — APPROVED** [review]
whoops, see I didn't LGTM since you updated 😊 😅

**Comment by thaJeztah** [comment]
Let's merge this; I see there's enough LGTM's, and we can update in follow-ups if desired 👍

**Pull request merged**
